app/controllers/application_controller.rb

```diff
@@ -1,11 +1,34 @@
+# frozen_string_literal: true
+
 class ApplicationController < ActionController::API
-  before_action :user_quota
+  around_action :set_time_zone, if: :current_user
+  before_action :user_quota, :api_limit
 
   def current_user
-    @current_user ||= User.find(request.headers["user-id"])
+    @current_user ||= User.includes(:hits).find(request.headers["user-id"])
   end
 
+  private
+
   def user_quota
-    render json: { error: 'over quota' } if current_user.count_hits >= 10000
+    render json: { error: 'over quota' } if blocked_user?
+  end
+
+  def blocked_user?
+    $redis.get(blocked_user_key).present?
```

Comment:
If the key exists on Redis, that means hits for that user are above the threshold.